### PR TITLE
fix(ci): update the Buf CI to push tags to the Buf Schema Registry

### PR DESCRIPTION
**Pull request by rach-id**
This PR fixes the push of a new tag to BSR. 

Tested in my fork in here: https://buf.build/chamirachid/nmt/labels

## Summary by CodeRabbit

- **New Features**
	- Introduced a new GitHub Actions workflow for continuous integration of Buf-related changes.
	- Added a workflow for automating the release of protobuf definitions to the Buf Schema Registry.
	- New script for downloading the specified version of the `buf` binary.

- **Bug Fixes**
	- Removed the outdated `buf-ci` workflow to streamline processes.

### CODE REVIEW

File: .github/workflows/buf-release.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: "push the tag label to BSR"
+        run: |
+          set -euo pipefail
+
+          bash ./scripts/buf-install.sh
+
+          echo ${{ secrets.BUF_TOKEN }} | buf registry login --token-stdin
+          
+          ./buf build
+          ./buf push --label ${{ github.ref_name }}
```
[the remark this answers is not in the document]
Comment (rach-id) [reply .github/workflows/buf-release.yml]: c4e7110

**Comment by rach-id** [comment]
failing go build CI is because I am creating the PR from a fork and codecov doesn't have permissions to check it.

the failing linter is fixed in https://github.com/celestiaorg/nmt/issues/278

the markdown contains some dead links but there is an issue to fix them

**Review by MSevey — APPROVED** [review]
utACK

**Review by evan-forbes — APPROVED** [review]

**Review by rootulp — COMMENTED** [review]
LGTM

**Pull request merged**